control-plane/pkg/reconciler/broker/broker.go

```diff
@@ -335,20 +335,23 @@ func (r *Reconciler) finalizeKind(ctx context.Context, broker *eventing.Broker)
 		return nil
 	}
 
-	_, externalTopic := isExternalTopic(broker)
 	secret, err := security.Secret(ctx, &security.MTConfigMapSecretLocator{ConfigMap: brokerConfig, UseNamespaceInConfigmap: false}, r.SecretProviderFunc())
 	if err != nil {
-
-		if externalTopic {
-			// we don't care
-			return nil
+		// If we can not get the referenced secret,
```

Comment:
regardless if the topic is externally managed or our own, let's see if we can try to get it, before we let it go on the finalization...